### ISSUE TITLE
Use OpenMP for updates

### ISSUE THREAD

**Issue by Alanocallaghan** [catavallejos/BASiCS#185]
Most are embarrassingly parallel

**Comment by Alanocallaghan** [comment]
With a toy example (100 cells, 1000 genes), wall clock time is roughly half with 4 cores. Speedup would be improved with more genes and/or more cells.

```r
> t0
   user  system elapsed 
108.013   0.104 108.154 
> t1
   user  system elapsed 
160.743   0.425  51.964 
```

**Comment by catavallejos** [comment]
This looks very exciting! Andreas said that he had some issues to run it in Mac, I can test on mine later this week.

**Comment by Alanocallaghan** [comment]
Yeah I gather OpenMP on OSX is tricky, my understanding is it doesn't come installed by default so might need installing with homebrew

**Comment by Alanocallaghan** [comment]
Could switch CI to github actions to get a better handle on the requirements (since it allows MacOS builds)
